packages/next/build/index.ts

```diff
@@ -2157,13 +2159,15 @@ export default async function build(
           }
 
           // Only move /404 to /404 when there is no custom 404 as in that case we don't know about the 404 page
-          if (!hasPages404 && useStatic404) {
-            await moveExportedPage('/_error', '/404', '/404', false, 'html')
-          }
 
-          if (useDefaultStatic500) {
-            await moveExportedPage('/_error', '/500', '/500', false, 'html')
-          }
+          await Promise.all([
+            ...(!hasPages404 && useStatic404
+              ? [moveExportedPage("/_error", "/404", "/404", false, "html")]
+              : []),
+            ...(useDefaultStatic500
+              ? [moveExportedPage("/_error", "/500", "/500", false, "html")]
+              : []),
+          ]);
 
```

Comment:
```suggestion
          await Promise.all([
            !hasPages404 && useStatic404 && moveExportedPage("/_error", "/404", "/404", false, "HTML"),
            useDefaultStatic500 && moveExportedPage("/_error", "/500", "/500", false, "html")
          ]);

```

We are not using the return value of `Promise.all`, so it doesn't require spread.

Reply:
Done :)